akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -11,11 +11,40 @@ import akka.http.scaladsl.server.AuthenticationFailedRejection._
 
 import scala.annotation.tailrec
 import scala.collection.immutable
+import scala.concurrent.ExecutionContext
 import scala.reflect.ClassTag
 
 trait RejectionHandler extends (immutable.Seq[Rejection] ⇒ Option[Route]) { self ⇒
   import RejectionHandler._
 
+  /** Map any HTTP response which was returned by this RejectionHandler to a different one before rendering it. */
+  def mapRejectionResponse(map: HttpResponse ⇒ HttpResponse)(implicit ec: ExecutionContext): RejectionHandler =
+    this match {
+      case a: BuiltRejectionHandler ⇒
+        new BuiltRejectionHandler(
+          a.cases.collect {
+            case ch: CaseHandler ⇒
+              CaseHandler(ch.pf.andThen(r ⇒ r.andThen(f ⇒ f.collect {
```

Comment:
Isn't that pattern just `mapResponse`? So, you should be able to write

```scala
ch.pf.andThen(mapResponse(map))
```

(and same for the below cases).